sirepo/pkcli/admin.py

```diff
@@ -141,9 +135,85 @@ def move_user_sims(target_uid=""):
         shutil.move(lib_file, target)
 
 
+def _build_ops(ops, simulations, sim_type, examples):
+    n = set([x.models.simulation.name for x in examples[sim_type]])
+    for sim in simulations:
+        if sim.name not in n:
+            ops.delete.append((sim, sim_type))
+        elif _example_is_too_old(sim.simulation.lastModified):
+            ops.revert.append((sim.name, sim_type))
+            ops.delete.append((sim, sim_type))
+
+
 def _create_example(example):
-    simulation_db.save_new_example(example)
+    try:
+        simulation_db.save_new_example(example)
+    except Exception as e:
+        pkdlog(
+            "Failed to create example: {} error={}", example.models.simulation.name, e
+        )
+
+
+def _delete(ops):
+    for s, t in ops.delete:
+        simulation_db.delete_simulation(t, s.simulationId)
+
+
+def _example_is_too_old(last_modified):
+    t = (srtime.utc_now_as_milliseconds() - last_modified) / _MILLISECONDS_PER_MONTH
+    if t > _MAXIMUM_SIM_AGE_IN_MONTHS:
+        return True
+    return False
+
+
+def _get_example_by_name(name, sim_type):
+    for example in simulation_db.examples(sim_type):
+        if example.models.simulation.name == name:
+            return example
+    raise AssertionError(f"Failed to find example simulation with name={name}")
+
+
+def _get_examples_by_type():
+    e = PKDict()
```

Comment:
I think a comprehension is simpler and more "pythonic"
`return PKDict({ t: simulation_db.examples(t) for t in feature_config.cfg().sim_types})`